### ISSUE TITLE
[Bug]: data race detected during bvt

### ISSUE THREAD

**Issue by sukki37** [matrixorigin/matrixone#6857]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93): b13440ebc66471642119070c503f652960f14610
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

It's too log to paste full log here, you can download by https://github.com/sukki37/matrixone/suites/9443470503/artifacts/446692749
```
==================
WARNING: DATA RACE
Read at 0x00c0475193c8 by goroutine 573799:
  github.com/matrixorigin/matrixone/pkg/container/vector.MustTCols[...]()
      /home/runner/work/matrixone/matrixone/head/pkg/container/vector/tools.go:52 +0x1d59
  github.com/matrixorigin/matrixone/pkg/vm/engine/disttae.(*Partition).NewReader()
      /home/runner/work/matrixone/matrixone/head/pkg/vm/engine/disttae/partition.go:485 +0x1c97
  github.com/matrixorigin/matrixone/pkg/vm/engine/disttae.(*table).newMergeReader()
      /home/runner/work/matrixone/matrixone/head/pkg/vm/engine/disttae/table.go:414 +0x986
  github.com/matrixorigin/matrixone/pkg/vm/engine/disttae.(*table).NewReader()
      /home/runner/work/matrixone/matrixone/head/pkg/vm/engine/disttae/table.go:318 +0x1aa
  github.com/matrixorigin/matrixone/pkg/sql/compile.(*Scope).ParallelRun()
      /home/runner/work/matrixone/matrixone/head/pkg/sql/compile/scope.go:169 +0x624
  github.com/matrixorigin/matrixone/pkg/sql/compile.(*Scope).RemoteRun()
      /home/runner/work/matrixone/matrixone/head/pkg/sql/compile/scope.go:129 +0x1ae
  github.com/matrixorigin/matrixone/pkg/sql/compile.(*Scope).MergeRun.func3()
      /home/runner/work/matrixone/matrixone/head/pkg/sql/compile/scope.go:90 +0xb7
  github.com/matrixorigin/matrixone/pkg/sql/compile.(*Scope).MergeRun.func8()
      /home/runner/work/matrixone/matrixone/head/pkg/sql/compile/scope.go:91 +0x47

Previous write at 0x00c0475193c8 by goroutine 2948:
  github.com/matrixorigin/matrixone/pkg/container/vector.New()
      /home/runner/work/matrixone/matrixone/head/pkg/container/vector/vector.go:508 +0xc69
  github.com/matrixorigin/matrixone/pkg/vm/engine/disttae.partitionDeleteBatch()
      /home/runner/work/matrixone/matrixone/head/pkg/vm/engine/disttae/tools.go:899 +0xc03
```

**Comment by aunjgr** [comment]
No progress today. Working on #6812

**Comment by m-schen** [comment]
I can't see what race it is. It seems that it's all read operations

**Comment by m-schen** [comment]
log:

Previous write at 0x00c0475193c8 by goroutine 2948:
  github.com/matrixorigin/matrixone/pkg/container/vector.New()
      /home/runner/work/matrixone/matrixone/head/pkg/container/vector/vector.go:508 +0xc69
  github.com/matrixorigin/matrixone/pkg/vm/engine/disttae.partitionDeleteBatch()
      /home/runner/work/matrixone/matrixone/head/pkg/vm/engine/disttae/tools.go:899 +0xc03
  github.com/matrixorigin/matrixone/pkg/vm/engine/disttae.(*table).Delete()
      /home/runner/work/matrixone/matrixone/head/pkg/vm/engine/disttae/table.go:288 +0x1e7
  github.com/matrixorigin/matrixone/pkg/sql/colexec/deletion.Call()
      /home/runner/work/matrixone/matrixone/head/pkg/sql/colexec/deletion/deletion.go:68 +0x6ec
  github.com/matrixorigin/matrixone/pkg/vm.Run()
      /home/runner/work/matrixone/matrixone/head/pkg/vm/vm.go:60 +0x20e
  github.com/matrixorigin/matrixone/pkg/vm/pipeline.(*Pipeline).MergeRun()
      /home/runner/work/matrixone/matrixone/head/pkg/vm/pipeline/pipeline.go:131 +0x238
  github.com/matrixorigin/matrixone/pkg/sql/compile.(*Scope).MergeRun()
      /home/runner/work/matrixone/matrixone/head/pkg/sql/compile/scope.go:111 +0xbf2
  github.com/matrixorigin/matrixone/pkg/sql/compile.(*Scope).Delete()
      /home/runner/work/matrixone/matrixone/head/pkg/sql/compile/dml.go:76 +0x552
  github.com/matrixorigin/matrixone/pkg/sql/compile.(*Compile).Run()
      /home/runner/work/matrixone/matrixone/head/pkg/sql/compile/compile.go:117 +0x4e4

**Comment by nnsgmsone** [comment]
may be fixed by #6931
